src/parsers/perf/perfparser.cpp

```diff
@@ -1009,6 +1012,37 @@ class PerfParserPrivate : public QObject
         }
     }
 
+    void addTracepointToTracepointStack(uint32_t tid, const Data::Tracepoint& tracepoint)
+    {
+        const auto tracepoints = Settings::instance()->tracepointParameters();
```

Comment:
this is not safe to be called from a background thread

generally, please add (in a separate patch) a `Q_ASSERT(QThread::currentThread() == settings.thread())` (and then compile with assertions enabled) to catch this in the future :)